cmd/operator/main.go

```diff
@@ -189,6 +197,26 @@ func init() {
 	flagset.StringVar(&cfg.AlertManagerSelector, "alertmanager-instance-selector", "", "Label selector to filter AlertManager Custom Resources to watch.")
 	flagset.StringVar(&cfg.ThanosRulerSelector, "thanos-ruler-instance-selector", "", "Label selector to filter ThanosRuler Custom Resources to watch.")
 	flagset.StringVar(&cfg.SecretListWatchSelector, "secret-field-selector", "", "Field selector to filter Secrets to watch")
+
+	// Prioritize specific flags over generic ones
+	// If config.CPU has been set but not config.CPURequest/config.CPULimit, default "" to config.CPU
+	// If config.Memory has been set but not config.MemoryRequest or config.MemoryLimit, default "" to config.Memory
+	if rcCPU != "" {
+		if cfg.ReloaderConfig.CPURequest == "" {
+			cfg.ReloaderConfig.CPURequest = rcCPU
+		}
+		if cfg.ReloaderConfig.CPULimit == "" {
+			cfg.ReloaderConfig.CPULimit = rcCPU
+		}
```

Comment:
I would log a deprecation warning if rcCPU is different than `100m` so folks using the flag are potentially aware of the deprecation.

Reply:
I can't address those then and there with logger so I baked the warning check into Main(). Let me know if there's a preferred alternative solution.